code/go/0chain.net/core/util/mpt_pnodedb_test.go

```diff
@@ -0,0 +1,339 @@
+				version:  tt.fields.version,
+				versions: tt.fields.versions,
+			}
+			if err := pndb.Iterate(tt.args.ctx, tt.args.handler); (err != nil) != tt.wantErr {
+				t.Errorf("Iterate() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestPNodeDB_PruneBelowVersion_Iterator_Err(t *testing.T) {
+	db, cleanUp := newPNodeDB(t)
+	defer cleanUp()
+
+	wo := gorocksdb.NewDefaultWriteOptions()
+
+	for i := 0; i < 257; i++ {
+		r := rand.Int()
+		err := db.db.Put(wo, []byte(strconv.Itoa(r)), []byte{NodeTypeValueNode})
+		require.NoError(t, err)
+	}
+
+	wo.DisableWAL(true)
+	wo.SetSync(true)
+
+	fo := gorocksdb.NewDefaultFlushOptions()
+	ctx := context.WithValue(context.TODO(), PruneStatsKey, &PruneStats{})
+
+	type fields struct {
+		dataDir  string
+		db       *gorocksdb.DB
+		ro       *gorocksdb.ReadOptions
+		wo       *gorocksdb.WriteOptions
+		to       *gorocksdb.TransactionOptions
+		fo       *gorocksdb.FlushOptions
+		version  int64
+		versions []int64
+	}
+	type args struct {
+		ctx     context.Context
+		version Sequence
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "Test_PNodeDB_PruneBelowVersion_OK",
+			fields:  fields{db: db.db, wo: wo, fo: fo},
+			args:    args{ctx: ctx, version: 1},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pndb := &PNodeDB{
+				dataDir:  tt.fields.dataDir,
+				db:       tt.fields.db,
+				ro:       tt.fields.ro,
+				wo:       tt.fields.wo,
+				to:       tt.fields.to,
+				fo:       tt.fields.fo,
+				version:  tt.fields.version,
+				versions: tt.fields.versions,
+			}
+			if err := pndb.PruneBelowVersion(tt.args.ctx, tt.args.version); (err != nil) != tt.wantErr {
+				t.Errorf("PruneBelowVersion() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestPNodeDB_PruneBelowVersion_Multi_Delete_Err(t *testing.T) {
+	db, cleanUp := newPNodeDB(t)
+	defer cleanUp()
+
+	wo := gorocksdb.NewDefaultWriteOptions()
+	n := NewValueNode()
+	err := db.db.Put(wo, []byte("key"), n.Encode())
+	require.NoError(t, err)
+
+	wo.DisableWAL(true)
+	wo.SetSync(true)
+
+	fo := gorocksdb.NewDefaultFlushOptions()
+	ctx := context.WithValue(context.TODO(), PruneStatsKey, &PruneStats{})
+
+	type fields struct {
+		dataDir  string
+		db       *gorocksdb.DB
+		ro       *gorocksdb.ReadOptions
+		wo       *gorocksdb.WriteOptions
+		to       *gorocksdb.TransactionOptions
+		fo       *gorocksdb.FlushOptions
+		version  int64
+		versions []int64
+	}
+	type args struct {
+		ctx     context.Context
+		version Sequence
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "Test_PNodeDB_PruneBelowVersion_OK",
+			fields:  fields{db: db.db, wo: wo, fo: fo},
+			args:    args{ctx: ctx, version: 1},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pndb := &PNodeDB{
```

Comment:
Move the fields defention inside the t.Run goroutine. The database needs to be setup inside the `t.Run` goroutine to avoid errors when used with t.Parallel.

Reply:
Done.